### PR TITLE
Cw tutorial/contract creation

### PR DESCRIPTION
**Pull request by jawoznia**

### CODE REVIEW

File: src/pages/tutorial/cw-contract/entry-points.mdx [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+#[entry_point]
+pub fn instantiate(_deps: DepsMut, _env: Env, _info: MessageInfo, _msg: Empty) -> StdResult<Response> {
+    Ok(Response::new())
+}
+```
+
+In fact, `instantiate` is the only entry point required for a smart contract to be valid. It is not
+very useful in this form, but it is a start. Let's take a closer look at the entry point structure.
+
+First, we start with importing couple of types just for more consistent usage. Then we define our
```
Comment (uint) [inline src/pages/tutorial/cw-contract/entry-points.mdx]:
```suggestion
First, we start with importing a couple of types just for more consistent usage. Then we define our
```

File: src/pages/tutorial/cw-contract/entry-points.mdx [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+entry point. The `instantiate` takes four arguments:
+
+- [`deps: DepsMut`](https://docs.rs/cosmwasm-std/latest/cosmwasm_std/struct.DepsMut.html) is a
+  utility type for communicating with the outer world - it allows querying and updating the contract
+  state, querying other contracts state, and gives access to an `Api` object with a couple of helper
+  functions for dealing with CW addresses.
+- [`env: Env`](https://docs.rs/cosmwasm-std/latest/cosmwasm_std/struct.Env.html) is an object
+  representing the blockchains state when executing the message - the chain height and id, current
+  timestamp, and the called contract address.
+- [`info: MessageInfo`](https://docs.rs/cosmwasm-std/latest/cosmwasm_std/struct.Env.html) contains
```
Comment (uint) [inline src/pages/tutorial/cw-contract/entry-points.mdx]:
```suggestion
- [`info: MessageInfo`](https://docs.rs/cosmwasm-std/latest/cosmwasm_std/struct.MessageInfo.html) contains
```

File: src/pages/tutorial/cw-contract/entry-points.mdx [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+  metainformation about the message which triggered an execution - an address that sends the
```
Comment (uint) [inline src/pages/tutorial/cw-contract/entry-points.mdx]:
```suggestion
  metainformation about the message that triggered the execution - an address that sends the
```

File: src/pages/tutorial/cw-contract/entry-points.mdx [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+  message, and chain native tokens sent with the message.
+- [`msg: Empty`](https://docs.rs/cosmwasm-std/latest/cosmwasm_std/struct.Empty.html) is the message
+  triggering execution itself - for now, it is `Empty` type that represents `{}` JSON, but the type
```
Comment (uint) [inline src/pages/tutorial/cw-contract/entry-points.mdx]:
```suggestion
  triggering execution itself - for now, it is the `Empty` type that represents `{}` JSON, but the type
```

File: src/pages/tutorial/cw-contract/entry-points.mdx [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+  of this argument can be anything that is deserializable, and we will pass more complex types here
+  in the future.
+
+If you are new to the blockchain, those arguments may not have much sense to you, but while
+progressing with this guide, I will explain their usage one by one.
```
Comment (uint) [inline src/pages/tutorial/cw-contract/entry-points.mdx]:
```suggestion
If you are new to the blockchain, those arguments may not make much sense to you, but as
you work through this guide, I will explain their usage one by one.
```

File: src/pages/tutorial/cw-contract/entry-points.mdx [hunk 6]
```diff
@@ -0,0 +1,72 @@
...
+
+Notice an essential attribute decorating our entry point
+[`#[entry_point]`](https://docs.rs/cosmwasm-std/latest/cosmwasm_std/struct.Empty.html). Its purpose
+is to wrap the whole entry point to the form Wasm runtime understands. The proper Wasm entry points
```
Comment (uint) [inline src/pages/tutorial/cw-contract/entry-points.mdx]:
```suggestion
is to wrap the whole entry point to the form Wasm runtime understands. Proper Wasm entry points
```

File: src/pages/tutorial/cw-contract/entry-points.mdx [hunk 7]
```diff
@@ -0,0 +1,72 @@
...
+- [`msg: Empty`](https://docs.rs/cosmwasm-std/latest/cosmwasm_std/struct.Empty.html) is the message
+  triggering execution itself - for now, it is `Empty` type that represents `{}` JSON, but the type
+  of this argument can be anything that is deserializable, and we will pass more complex types here
+  in the future.
+
+If you are new to the blockchain, those arguments may not have much sense to you, but while
+progressing with this guide, I will explain their usage one by one.
+
+Notice an essential attribute decorating our entry point
+[`#[entry_point]`](https://docs.rs/cosmwasm-std/latest/cosmwasm_std/struct.Empty.html). Its purpose
```
Comment (uint) [inline src/pages/tutorial/cw-contract/entry-points.mdx]:
```suggestion
[`#[entry_point]`](https://docs.rs/cosmwasm-std/latest/cosmwasm_std/attr.entry_point.html). Its purpose
```

File: src/pages/tutorial/cw-contract/entry-points.mdx [hunk 8]
```diff
@@ -0,0 +1,72 @@
...
+creators delivered the entry_point macro. It creates the raw Wasm entry point, calling the decorated
+function internally and doing all the magic required to build our high-level Rust arguments from
+arguments passed by Wasm runtime.
+
+The next thing to look at is the return type. I used
+[`StdResult<Response>`](https://docs.rs/cosmwasm-std/latest/cosmwasm_std/type.StdResult.html) for
+this simple example, which is an alias for `Result<Response, StdError>`. The return entry point type
+would always be a [`Result`](https://doc.rust-lang.org/std/result/enum.Result.html) type, with some
+error type implementing [`ToString`](https://doc.rust-lang.org/std/string/trait.ToString.html) trait
+and a well-defined type for success case. For most entry points, an "Ok" case would be the
```
Comment (uint) [inline src/pages/tutorial/cw-contract/entry-points.mdx]:
```suggestion
and a well-defined type for the success case. For most entry points, an "Ok" case would be the
```

**Review by uint — APPROVED** [review]
Overall, a nice, user-friendly guide that leaves no mysteries.

I cannot do this without nitpicks, apparently ;)

File: src/pages/tutorial/cw-contract/entry-points.mdx (hunk 2 above)
Comment (jawoznia) [reply src/pages/tutorial/cw-contract/entry-points.mdx]: Great catch!

**Pull request merged**